api/helpers/game-states/moves/seven-scuttle/validate.js

```diff
@@ -0,0 +1,68 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Validate request to seven-scuttle',
+
+  description: 'Verifies whether a request to scuttle with a seven is legal, throwing an explanatory error if not.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'Object containing the current game state',
+      required: true,
+    },
+    /**
+     * @param { Object } requestedMove - Object describing the request to scuttle from the top of the deck
+     * @param { String } requestedMove.cardId - Card Played for scuttle
+     * @param { String } requestedMove.targetId - Card Targeted for scuttle
+     * @param { MoveType.SEVEN_SCUTTLE } requestedMove.moveType - Specifies that this a sevenScuttle
+     */
+    requestedMove: {
+      type: 'ref',
+      description: 'Object describing the request to scuttle',
+      required: true,
+    },
+    playedBy: {
+      type: 'number',
+      description: 'Player number of player requesting the move',
+      required: true,
+    },
+  },
+
+  sync: true,
+
+  fn: ({ requestedMove, currentState, playedBy }, exits) => {
+    try {
+      const opponent = playedBy ? currentState.p0 : currentState.p1;
+
+      const topTwoCards = currentState.deck.slice(0, 2);
+      const targetCard = opponent.points.find(({ id }) => id === requestedMove.targetId);
+      const playedCard = topTwoCards.find(({ id }) => id === requestedMove.cardId);
+      const cardPlayedIndex = topTwoCards.findIndex(({ id }) => id === requestedMove.cardId);
```

Comment:
Nitpick: we don't need the index and can just use playedCard, since we aren't doing any array manipulation in `validate()`. The `execute()` helpers use `.findIndex()` so that they can call `.splice()` to actually move cards around.

So keep line 40 as it is and remove 41
```suggestion
      const playedCard = topTwoCards.find(({ id }) => id === requestedMove.cardId);
```